### ISSUE TITLE
Update multi-instrumentation Readme for 0.110.0

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-operator#3341]
### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-operator/releases/tag/v0.110.0 enables multi-instrumentation by default readme should be updated to reflect the change.

e.g. should be fixed
> NOTE: instrumentation.opentelemetry.io/container-names annotation is not used for this feature.

**Comment by pavolloffay** [comment]
is there anything else we should udpate?

**Comment by iblancasa** [comment]
I don't think so since enabling the multi-instrumentation is part of the changelog.

**Comment by pavolloffay** [comment]
> Works only when enable-multi-instrumentation flag is true.

https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#multi-container-pods-with-multiple-instrumentations 

Perhaps we should indicate that it is enabled by default and `instrumentation.opentelemetry.io/container-names` continues to function correctly
